### PR TITLE
[6971] Add Rake Task to Cleanup Stale Users

### PR DESCRIPTION
**Pull request by AdColvin**
### Context
[Trello](https://trello.com/c/lA8GyJm7)

[It's been found](https://educationgovuk.sharepoint.com.mcas.ms/:x:/r/sites/TeacherServices/_layouts/15/doc2.aspx?sourcedoc=%7BFCACCD27-B893-4ED0-A212-E37FAABBE6AF%7D&file=user_activity_2024.xlsx&action=default&mobileredirect=true) that we have several hundred Trainees in register that are considered inactive, because they haven't signed in for over a year, or they've never signed in at all.

### Changes proposed in this pull request
This adds a rake task that can be given a text file full of emails addresses and it will discard (soft delete) them. There's a small chance that some users will want their account reinstated so soft deletes are preferred.

Once this is merged, I can create the text file with the email addresses on production and run the rake task with the file from there. 

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: lib/tasks/cleanup.rake [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+namespace :cleanup do
+  desc "Destroy trainees based on email list"
+  task :discard_trainees, [:file_path] => :environment do |_t, args|
+    raise "You must provide a file path" unless args[:file_path]
+
+    emails = File.readlines(args[:file_path]).map(&:chomp)
+    trainees = Trainee.where(email: emails)
+    trainees.discard_all
```
Comment (Reyko) [inline lib/tasks/cleanup.rake:10]: How about adding the variation which will raise an Exception if anything goes wrong? [discard_all!](https://github.com/jhawthorn/discard/blob/c52de70bca2521a193f85dcb2424afa89074bddc/lib/discard/model.rb#L61)

```suggestion
    trainees.discard_all!
```

File: spec/lib/tasks/cleanup_trainees_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "rake"
+
+describe "cleanup:discard_trainees" do
```
Comment (Reyko) [inline spec/lib/tasks/cleanup_trainees_spec.rb:6]: 👍

File: spec/lib/tasks/cleanup_trainees_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  context "when file path is provided" do
+    let(:file_path) { Rails.root.join("spec/fixtures/files/cleanup_emails.txt") }
+    let(:emails) { File.readlines(file_path).map(&:chomp) }
+
+    before do
+      emails.each { |email| create(:trainee, email:) }
+    end
+
+    it "destroys the trainees with emails in the file" do
```
Comment (Reyko) [inline spec/lib/tasks/cleanup_trainees_spec.rb:25]: Nitpick: 

```suggestion
    it "soft deletes the trainees with emails in the file" do
```

**Comment by AdColvin** [comment]
Thanks for the review @Reyko, I committed your suggestions and looks like we have a green build 👍

**Review by Reyko — APPROVED** [review]

**Comment by AdColvin** [comment]
@stevehook @Reyko Tom pointed out to me that we actually need to be discarding users and not trainees. Good job he checked! I've updated the task to do that and also made sure that any blank lines are discounted before running the query. That should address your concern about deleting users with null emails Steve. Let me know what you think.

**Review by Reyko — APPROVED** [review]

**Pull request merged**